src/Npgsql/TypeMapping/GlobalTypeMapper.cs

```diff
@@ -88,14 +76,41 @@ public override INpgsqlTypeMapper AddMapping(NpgsqlTypeMapping mapping)
             }
         }
 
+        void UpdateNonMappingTables(NpgsqlTypeMapping mapping)
+        {
+            if (mapping.NpgsqlDbType.HasValue)
+            {
+                _npgsqlDbTypeToPgTypeName[mapping.NpgsqlDbType.Value] = mapping.PgTypeName;
+                _npgsqlDbTypeToPgTypeName[mapping.NpgsqlDbType.Value | NpgsqlDbType.Array] = mapping.PgTypeName + "[]";
+
+                foreach (var dbType in mapping.DbTypes)
+                    _dbTypeToNpgsqlDbType[dbType] = mapping.NpgsqlDbType.Value;
+
+                if (mapping.InferredDbType.HasValue)
+                    _npgsqlDbTypeToDbType[mapping.NpgsqlDbType.Value] = mapping.InferredDbType.Value;
+
+                foreach (var clrType in mapping.ClrTypes)
+                {
+                    _typeToNpgsqlDbType[clrType] = mapping.NpgsqlDbType.Value;
+                    _typeToPgTypeName[clrType] = mapping.PgTypeName;
+                }
+            }
+
+            if (mapping.InferredDbType.HasValue)
+                foreach (var clrType in mapping.ClrTypes)
+                    _typeToDbType[clrType] = mapping.InferredDbType.Value;
+        }
+
         public override bool RemoveMapping(string pgTypeName)
         {
             Lock.EnterWriteLock();
             try
             {
-                var result = base.RemoveMapping(pgTypeName);
+                var oldMappingsByName = MappingsByName;
+                MappingsByName = MappingsByName.Remove(pgTypeName);
+                var changed = ReferenceEquals(MappingsByName, oldMappingsByName);
```

Comment:
Pretty sure it's going to be `true` if there is no changes, and `false` otherwise. Which means, we return `false` even if we did remove the mapping.

If so, should we write some tests for this behavior?

Reply:
Good catch, the logic is inversed. I will add a test.